### PR TITLE
KOGITO-3325 DMN hot reload manage accessor cache when stronglytyped

### PR DESCRIPTION
**Pull request by tarilabs**
https://issues.redhat.com/browse/KOGITO-3325

### CODE REVIEW

File: kogito-quarkus-extension/integration-test-hot-reload/src/test/java/io/quarkus/it/kogito/decision/SimpleModifyHotReloadTest.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            new Supplier<JavaArchive>() {
+
+                @Override
+                public JavaArchive get() {
+                    JavaArchive ja = ShrinkWrap.create(JavaArchive.class)
+                            .addAsResource("application.properties.stronglytyped", "/application.properties")
+                            .addAsResource("TrafficViolation.txt", DMN_RESOURCE_FILE);
+                    return ja;
+                }
+            });
```
Comment (danielezonca) [inline kogito-quarkus-extension/integration-test-hot-reload/src/test/java/io/quarkus/it/kogito/decision/SimpleModifyHotReloadTest.java:56]:
```suggestion
    final static QuarkusDevModeTest test = new QuarkusDevModeTest().setArchiveProducer(
            () -> ShrinkWrap.create(JavaArchive.class)
                            .addAsResource("application.properties.stronglytyped", "/application.properties")
                            .addAsResource("TrafficViolation.txt", DMN_RESOURCE_FILE));
```

File: drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DMNKogito.java [hunk 2]
```diff
@@ -60,6 +61,7 @@ public static DMNRuntime createGenericDMNRuntime(Reader... readers) {
 
     public static DMNRuntime createGenericDMNRuntime(Function<String, KieRuntimeFactory> kiePMMLRuntimeFactoryFunction, Reader... readers) {
         List<Resource> resources = Stream.of(readers).map(ReaderResource::new).collect(Collectors.toList());
+        EvalHelper.clearGenericAccessorCache();
```
Comment (danielezonca) [inline drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DMNKogito.java:64]: What about a comment to explain this code?

**Review by danielezonca — APPROVED** [review]
LGTM :+1: 
Two minor optional comments

**Review by hellowdan — APPROVED** [review]

File: kogito-quarkus-extension/integration-test-hot-reload/src/test/java/io/quarkus/it/kogito/decision/SimpleModifyHotReloadTest.java (hunk 1 above)
Comment (tarilabs) [reply kogito-quarkus-extension/integration-test-hot-reload/src/test/java/io/quarkus/it/kogito/decision/SimpleModifyHotReloadTest.java:56]: rejected, lambda here is useless, we do Quarkus style which is anonymous class ;)

File: drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DMNKogito.java (hunk 2 above)
Comment (tarilabs) [reply drools/kogito-dmn/src/main/java/org/kie/kogito/dmn/DMNKogito.java]: done 👍

**Comment by tarilabs** [comment]
image

```
Existing failures - 1
testPingPongBetweenProcessInstances – org.kie.kogito.integrationtests.springboot.PingPongMessageTest
8s
Error
Assertion condition defined as a lambda expression in org.kie.kogito.integrationtests.springboot.PingPongMessageTest that uses java.lang.String 1 expectation failed.
JSON path message doesn't match.
Expected: hello world
Actual: hello
within 5 seconds.
Stacktrace
org.awaitility.core.ConditionTimeoutException:
Assertion condition defined as a lambda expression in org.kie.kogito.integrationtests.springboot.PingPongMessageTest that uses java.lang.String 1 expectation failed.
JSON path message doesn't match.
Expected: hello world
Actual: hello
within 5 seconds.
at org.kie.kogito.integrationtests.springboot.PingPongMessageTest.testPingPongBetweenProcessInstances(PingPongMessageTest.java:53)
Caused by: java.lang.AssertionError:
1 expectation failed.
JSON path message doesn't match.
Expected: hello world
Actual: hello
at org.kie.kogito.integrationtests.springboot.PingPongMessageTest.lambda$testPingPongBetweenProcessInstances$0(PingPongMessageTest.java:59)
```

Jenkins retest this

**Review by mariofusco — APPROVED** [review]

**Pull request merged**